### PR TITLE
[GraphBolt] update FeatureStore to use tuple as key

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/feature_store.py [hunk 1]
```diff
@@ -142,21 +142,29 @@ def update(self, value: torch.Tensor, ids: torch.Tensor = None):
             self._tensor[ids] = value
 
 
-# FIXME(Rui): To avoid circular import, we make a copy of `OnDiskDataFormatEnum`
-# from dataset.py. We need to merge the two definitions later.
-class OnDiskDataFormatEnum(pydantic_yaml.YamlStrEnum):
-    """Enum of data format."""
+class FeatureDataFormatEnum(pydantic_yaml.YamlStrEnum):
```
Comment (frozenbugs) [inline python/dgl/graphbolt/feature_store.py:145]: Merge the whole implementation here to python/dgl/graphbolt/impl/ondisk_dataset.py.

Comment (Rhett-Ying) [reply python/dgl/graphbolt/feature_store.py:145]: `OnDiskDataset` should use the class defined here as `feature_store` is a dependency of `OnDiskDataset`.

Comment (frozenbugs) [reply python/dgl/graphbolt/feature_store.py:145]: There are several APIs defined in graphbot/, and all the implementation should go into graphbot/impl. We can put all OnDiskRelated loading code to a single python file, and it depends on related API defined in graphbot/.

Comment (Rhett-Ying) [reply python/dgl/graphbolt/feature_store.py:145]: sure. I added a `TODO` comment ahead this line and I'd like to finish it in another separate PR.

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**